asapdiscovery-modeling/asapdiscovery/modeling/modeling.py

```diff
@@ -0,0 +1,132 @@
+from asapdiscovery.data.openeye import oechem, oespruce
+
+
+def spruce_protein(
+    initial_prot: oechem.OEGraphMol,
+    return_du=False,
+    seqres: str = None,
+    loop_db: str = None,
+    site_residue="HIS:41: :A:0: ",
+) -> oechem.OEDesignUnit or oechem.OEGraphMol:
+    """
+    Applies the OESpruce protein preparation pipeline to the given protein structure.
+
+    Args:
+    - initial_prot (oechem.OEMol): the input protein structure to be prepared.
+    - return_du (bool, optional): whether to return a design unit (DU) as the output. If True, the function will
+      return the DU generated by the OESpruce pipeline; if False, the function will return the prepared protein
+      structure as an oechem.OEGraphMol. Default is False.
+    - seqres (str, optional): the SEQRES string of the protein. If provided, the SEQRES metadata will be added
+      to the structure before applying the OESpruce pipeline. Default is None.
+    - loop_db (str, optional): the filename of the loop database to be used by the OESpruce pipeline. If provided,
+      the pipeline will include loop building step. Default is None.
+    - site_residue (str, optional): the site residue used to define the binding site of the protein. Default is
+      "HIS:41: :A:0: ". This is necessary when there is no ligand in the input structure, otherwise OpenEye will not
+      know where to put the binding site.
+
+    Returns:
+    - oechem.OEMol: the prepared protein structure, or a DU generated by the OESpruce pipeline if return_du is True.
+      If design unit preparation fails, it will return the prepared protein structure instead.
+    """
```

Comment:
(Not a blocker) I can tell ChatGPT was used for this :) haha. It is fine, but we probably want to change the format of the docstring to meet the one we commonly use in the lab, that is using the numpy format. You can probably ask ChatGPT to format it that way.

Reply:
Yeah it was so nice :P maybe there's an automatic documentation formatter?